crates/flipperzero/src/furi/sync.rs

```diff
@@ -17,16 +18,13 @@ type UnsendUnsync = PhantomData<*const ()>;
 
 /// A mutual exclusion primitive useful for protecting shared data.
 pub struct Mutex<T: ?Sized> {
-    mutex: *mut sys::FuriMutex,
+    mutex: NonNull<sys::FuriMutex>,
     data: UnsafeCell<T>,
 }
 
 impl<T> Mutex<T> {
     pub fn new(data: T) -> Self {
-        let mutex = unsafe { sys::furi_mutex_alloc(MUTEX_TYPE) };
-        if mutex.is_null() {
-            panic!("furi_mutex_alloc failed");
-        }
+        let mutex = unsafe { NonNull::new_unchecked(sys::furi_mutex_alloc(MUTEX_TYPE)) };
```

Comment:
This calls either `xSemaphoreCreateMutex` or `xSemaphoreCreateRecursiveMutex`, and then uses `furi_check` to enforce non-null.